runbot/models/branch.py

```diff
@@ -87,6 +87,28 @@ def _compute_reference_name(self):
                 reference_name = f'{forced_version.name}---{reference_name}'
             branch.reference_name = reference_name
 
+    def _match_errors_from_body(self):
+        """
+        Checks the pr_body for any reference to runbot errors and links the branch
+        to those errors if they exist.
+        """
+        self.ensure_one()
+        if not self.pr_body:
+            return
+        pr_body_regex = r'^.*runbot.*error[^\d]*(\d+)[ \S]*\r?$'
+        error_ids = re.findall(
+            pr_body_regex,
+            self.pr_body,
+            re.IGNORECASE | re.MULTILINE
+        )
```

Comment:
I think the regex can be simplified to this:
```suggestion
        pr_body_regex = 
        error_ids = re.findall(
            r'\brunbot.*error.*?(\d+)',
            self.pr_body,
            re.IGNORECASE | re.ASCII
        )
```
- there's no reason to limit the pattern to one match per line
- `\d+` is greedy so there's no need to end it with something wonky, especially not "space or non-whitespace"
- no idea why `\r` is even there
- by default re will consider e.g. `២` to be a valid `\d`, which probably isn't desirable here